### PR TITLE
Feature/splay tree

### PR DESCRIPTION
**Pull request by Snover98**
check if there are any problems in the finished trees

### CODE REVIEW

File: binTree.h [hunk 1]
```diff
@@ -0,0 +1,391 @@
+#ifndef DS_WET_2_BINTREE_H
+#define DS_WET_2_BINTREE_H
+
+#include <iostream>
```
Comment (Snover98) [inline binTree.h]: remove includes in lines 4 and 5

File: binTree.h [hunk 2]
```diff
@@ -0,0 +1,391 @@
...
+    TreeNode* root;
+    Compare comp;
+
+    TreeNode* findNode(T& info){
+        TreeNode* curr = root;
+
+        while(curr != NULL) {
+            //if this is the node
+            if(comp(info, curr->info) == 0) {
+                return (curr->info);
```
Comment (raoded) [inline binTree.h]: needs to be just 'return curr' instead of curr->info

File: binTree.h [hunk 3]
```diff
@@ -0,0 +1,391 @@
...
+
+    template <typename Func>
+    void Postorder(Func func);
+};
+
+template<class T, class Compare>
+T& BinTree<T, Compare>::find(T& info) {
+    //search for a node with the info
+    TreeNode* found = findNode(info);
+
```
Comment (Snover98) [inline binTree.h]: check if found is NULL

File: binTree.h [hunk 4]
```diff
@@ -0,0 +1,391 @@
...
+        return found->info;
+    }
+
+    //otherwise, there is no node with the info
+    return NULL;
+}
+
+template<class T, class Compare>
+T& BinTree<T, Compare>::findMax() {
+    return findMaxNode()->info;
```
Comment (Snover98) [inline binTree.h]: check if it's NULL

File: binTree.h [hunk 5]
```diff
@@ -0,0 +1,391 @@
...
+}
+
+template<class T, class Compare>
+T& BinTree<T, Compare>::findMin() {
+    return findMinNode()->info;
```
Comment (Snover98) [inline binTree.h]: check if it's NULL

File: binTree.h [hunk 6]
```diff
@@ -0,0 +1,391 @@
...
+    t->left = NULL;
+    t->right = NULL;
+    t->parent = NULL;
+
+    //If this is a new tree
+    if(isEmpty()) {
+        root = t;
+    }
+
+    TreeNode* new_parent = root;
```
Comment (Snover98) [inline binTree.h]: you can use findNode to find instead

File: binTree.h [hunk 7]
```diff
@@ -0,0 +1,391 @@
...
+    t->parent = new_parent;
+}
+
+template<class T, class Compare>
+bool BinTree<T, Compare>::remove(T& info) {
+    if(isEmpty()) {
+        return false;
+    }
+
+    TreeNode* curr = find(info);
```
Comment (Snover98) [inline binTree.h]: should be findNode

File: binTree.h [hunk 8]
```diff
@@ -0,0 +1,391 @@
...
+
+    //if there is no node with that info in the tree
```
Comment (Snover98) [inline binTree.h]: check pointer games

**Pull request merged**